app.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require_relative 'models'
+
+get '/' do
+  redirect '/memos'
+end
+
+get '/memos' do
+  @title = 'memos'
+  @memos = parse_all_data
+  erb :index
+end
+
+get '/memos/new' do
+  @title = 'new'
+  erb :new
+end
+
+post '/memos' do
+  memo = Memo.new(name: params[:name], text: params[:text])
+  memo.save_new_memo
+  redirect '/memos'
+end
+
+get '/memos/:id/edit' do
+  @title = 'edit'
+  @id = params[:id].to_i
+  find_memo_data
```

Comment:
viewに対してはインスタンス変数で値を受け渡す方法しかありませんが、それ以外は引数で渡すようにしましょう。
そうしないとプログラムが大きくなってきたときにカオスになって収拾が付かなくなります。

```suggestion
  id = params[:id].to_i
  find_memo_data(id)
```

同様に、`find_memo_data`も戻り値を返して、それをapp.rbでインスタンス変数にセットしてください。
あっちこっちでインスタンス変数にセットすると、問題が起きたときにデバッグが難しくなります。

[改訂版・（あなたの周りでも見かけるかもしれない）インスタンス変数の間違った使い方 \- Qiita](https://qiita.com/jnchito/items/4d62693525f5023013cc)

Reply:
記事を読んで、間違った使い方をもろにしていたことに気づきました💦
あっちこっちのコインロッカーにデータを入れていて、自分でもよくわからなくなっていました。

「メソッドに引数で値を渡す → メソッドの戻り値にハッシュを返す → ハッシュから値を取り出して、インスタンス変数にセットする」という流れで書き直しました。https://github.com/SuzukaHori/sinatra_practice/pull/1/commits/d1e6038a4c5cb288078278d6d5053ebd51d56ef3